libfastpm/solver.c

```diff
@@ -390,14 +382,16 @@ fastpm_do_force(FastPMSolver * fastpm, FastPMTransition * trans)
 
     FastPMForceEvent event[1];
 
-    /* FIXME modify gravity.c to compute delta_k from all species.
-       [ Only CDM is used to calc N below ] */
+    /* FIXME modify gravity.c to compute delta_k from all species. */
     FastPMStore * p = fastpm_solver_get_species(fastpm, FASTPM_SPECIES_CDM);
+    int64_t N = p->np;
+    if (fastpm_solver_get_species(fastpm, FASTPM_SPECIES_NCDM)) {
+        FastPMStore * ncdm = fastpm_solver_get_species(fastpm, FASTPM_SPECIES_NCDM);
+        N += ncdm->np;
+    }
```

Comment:
@rainwoodman Is this change necessary for ncdm? Do we need to include ncdm->np in N?

Reply:
From what I see this is only ever used when outputting the shot noise, N1 and N2 when writing the powerspectrum, so it won't affect the functionality of the code.

Reply:
I don't think we need to include N of ncdm to the shotnoise of the total CDM -- my gut feeling is that ncdm shouldn't have changed the shotnoise level due to the mass weighting. Perhaps remove this N += line, and add a comment why we think it is a good idea to neglect ncdm contribution to shotnoise of the total power spectrum.

Reply:
Okay thanks, will do. Just to check, am I correct to say this won't affect the results of the snapshots -- it only affects powerspec.txt?